src/Searchable.php

```diff
@@ -26,38 +27,59 @@
  */
 class Searchable extends DataExtension
 {
-    public static $published_field = 'SS_Published';
+    public const TYPE_FIELD = 'type';
+
+    public const PUBLISHED_FIELD = 'SS_Published';
 
     /**
      * @config
      * @var array
      */
-    public static $mappings = array(
+    private static $elasticsearch_field_mappings = [
         'PrimaryKey'  => 'integer',
         'ForeignKey'  => 'integer',
-        'DBClassName' => 'string',
+        'DBClassName' => 'keyword',
         'DBDatetime'  => 'date',
```

Comment:
Does Es' internal `date` field comprise "time" as well therefore?

Reply:
looks like it: https://www.elastic.co/guide/en/elasticsearch/reference/7.16/date.html